src/django_upgrade/fixers/empty_result_set_exception.py

```diff
@@ -0,0 +1,63 @@
+"""
+Replace compatibility imports for django.core.exceptions.EmptyResultSet:
+https://docs.djangoproject.com/en/3.1/releases/3.1/#id1
+"""
+import ast
+from collections import defaultdict
+from functools import partial
+from typing import Iterable, List, Tuple
+
+from tokenize_rt import Offset, Token
+
+from django_upgrade.ast import ast_start_offset
+from django_upgrade.data import Fixer, State, TokenFunc
+from django_upgrade.tokens import extract_indent, insert, update_imports
+
+fixer = Fixer(
+    __name__,
+    min_version=(1, 11),
+)
+
+REWRITES = {
+    "django.db.models.query": {
+        "EmptyResultSet": "django.core.exceptions",
+    },
+    "django.db.models.sql": {
+        "EmptyResultSet": "django.core.exceptions",
+    },
+    "django.db.models.sql.datastructures": {
+        "EmptyResultSet": "django.core.exceptions",
+    },
+}
+
+
+@fixer.register(ast.ImportFrom)
+def visit_ImportFrom(
+    state: State,
+    node: ast.ImportFrom,
+    parent: ast.AST,
+) -> Iterable[Tuple[Offset, TokenFunc]]:
+    if (
+        node.level == 0
+        and node.module in REWRITES
+        and any(alias.name in REWRITES[node.module] for alias in node.names)
+    ):
+        yield ast_start_offset(node), partial(fix_import, node=node)
+
+
+def fix_import(tokens: List[Token], i: int, *, node: ast.ImportFrom) -> None:
```

Comment:
This is a copy of the existing function from `jsonfield.py`:

https://github.com/adamchainz/django-upgrade/blob/d1ceeee6f5e9bc447830c27aa2117e8beb24df7b/src/django_upgrade/fixers/jsonfield.py#L55

Feels like it could be moved into `django_upgrade.tokens`, renamed to `rewrite_imports` and shared between the 2 modules.